server/block/jukebox.go

```diff
@@ -0,0 +1,103 @@
+package block
+
+import (
+	"fmt"
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/entity"
+	"github.com/df-mc/dragonfly/server/internal/nbtconv"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+	"time"
+)
+
+// Jukebox is a block used to play music discs.
+type Jukebox struct {
+	solid
+	bass
+
+	// Item is the disc that is displayed inside the frame.
+	Item item.Stack
+}
+
+// FuelInfo ...
+func (j Jukebox) FuelInfo() item.FuelInfo {
+	return newFuelInfo(time.Second * 15)
+}
+
+// BreakInfo ...
+func (j Jukebox) BreakInfo() BreakInfo {
+	d := []item.Stack{item.NewStack(Jukebox{}, 1)}
+	if !j.Item.Empty() {
+		d = append(d, j.Item)
+	}
+	return newBreakInfo(2, alwaysHarvestable, axeEffective, simpleDrops(d...))
+}
+
+// jukeboxUser represents an item.User that can use a jukebox.
+type jukeboxUser interface {
+	item.User
+	// SendJukeboxPopup sends a jukebox popup to the item.User
+	SendJukeboxPopup(a ...any)
+}
+
+// Activate ...
+func (j Jukebox) Activate(pos cube.Pos, _ cube.Face, w *world.World, u item.User) bool {
+	if !j.Item.Empty() {
+		ent := entity.NewItem(j.Item, pos.Side(cube.FaceUp).Vec3Middle())
+		ent.SetVelocity(mgl64.Vec3{rand.Float64()*0.2 - 0.1, 0.2, rand.Float64()*0.2 - 0.1})
+		w.PlaySound(pos.Vec3(), sound.MusicDiscEnd{})
+		w.AddEntity(ent)
+
+		j.Item = item.Stack{}
+	} else if held, other := u.HeldItems(); !held.Empty() {
+		if m, ok := held.Item().(item.MusicDisc); ok {
+			j.Item = item.NewStack(held.Item(), 1)
+
+			u.SetHeldItems(held.Grow(-1), other)
```

Comment:
This does not account for creative players

Reply:
Neither do other blocks, like item frames

Reply:
These are creative parity-issues that are out of scope for this pr

Reply:
Can you please open an issue for this separate to this PR?